ambry-rest/src/main/java/com.github.ambry.rest/NettyServerFactory.java

```diff
@@ -47,17 +52,24 @@
    * @param sslFactory the {@link SSLFactory} used to construct the {@link javax.net.ssl.SSLEngine} used for handling
    *                   SSL requests.
    * @throws IllegalArgumentException if any of the arguments are null.
+   * @throws ReflectiveOperationException if a netty-specific {@link SSLFactory} cannot be instantiated via reflection.
    */
   public NettyServerFactory(VerifiableProperties verifiableProperties, MetricRegistry metricRegistry,
       final RestRequestHandler requestHandler, final PublicAccessLogger publicAccessLogger,
-      final RestServerState restServerState, SSLFactory sslFactory) {
+      final RestServerState restServerState, SSLFactory sslFactory) throws ReflectiveOperationException {
     if (verifiableProperties == null || metricRegistry == null || requestHandler == null || publicAccessLogger == null
         || restServerState == null) {
       throw new IllegalArgumentException("Null arg(s) received during instantiation of NettyServerFactory");
     }
     nettyConfig = new NettyConfig(verifiableProperties);
-    if (sslFactory == null && nettyConfig.nettyServerEnableSSL) {
-      throw new IllegalArgumentException("NettyServer requires SSL, but sslFactory is null");
+    if (nettyConfig.nettyServerEnableSSL) {
+      if (!nettyConfig.nettyServerSslFactory.isEmpty()) {
+        LOGGER.info("Using " + nettyConfig.nettyServerSslFactory + " for Netty SSL instead of the shared instance.");
+        sslFactory = Utils.getObj(nettyConfig.nettyServerSslFactory, new SSLConfig(verifiableProperties));
+      }
+      if (sslFactory == null) {
+        throw new IllegalArgumentException("NettyServer requires SSL, but sslFactory is null");
```

Comment:
Sorry, perhaps I should have created a new variable instead of overwriting the constructor argument. The provided constructor argument is the "default" sslFactory that is shared with the router and is constructed at a higher level. If the "netty.server.ssl.factory" option is set, then line 68 will ensure that netty constructs and uses "its own" factory.

Reply:
ok thanks for explaining.  Maybe could be documented a bit clearer.